examples/tests/valid/inflight_ref_resource_collection.w

```diff
@@ -0,0 +1,16 @@
+bring cloud;
+
+resource Test {
+  array: Array<cloud.Bucket>;
+
+  init() {
+    this.array = [new cloud.Bucket()];
+  }
+
+  inflight test() {
+    assert(this.array.at(0).list().length == 0);
```

Comment:
We don't know which operation is performed on the contents of the array, and I am planning to make this fail as part of https://github.com/winglang/wing/pull/1610

So for now, this is will only fail at runtime (not ideal, but I am on it).